File: Aliucord/src/main/java/com/aliucord/wrappers/messages/AttachmentWrapper.java
/*
 * Copyright (c) 2021 Juby210
 * Licensed under the Open Software License version 3.0
 */

package com.aliucord.wrappers.messages;

import com.discord.api.message.attachment.MessageAttachment;
import com.discord.api.message.attachment.MessageAttachmentType;

/**
 * Wraps the obfuscated {@link MessageAttachment} class to provide nice method names and require only one central
 * update if method names change after an update
 */
@SuppressWarnings({"unused", "deprecation"})
public class AttachmentWrapper {
    private final MessageAttachment attachment;

    public AttachmentWrapper(MessageAttachment attachment) {
        this.attachment = attachment;
    }

    public final String getFilename() {
        return getFilename(attachment);
    }

    public final Integer getHeight() {
        return getHeight(attachment);
    }

    public final String getProxyUrl() {
        return getProxyUrl(attachment);
    }

    public final long getSize() {
        return getSize(attachment);
    }

    public final MessageAttachmentType getType() {
        return getType(attachment);
    }

    public final String getUrl() {
        return getUrl(attachment);
    }

    public final Integer getWidth() {
        return getWidth(attachment);
    }



    public static String getFilename(MessageAttachment attachment) {
        return attachment.a();
    }

    public static Integer getHeight(MessageAttachment attachment) {
        return attachment.b();
    }

    public static String getProxyUrl(MessageAttachment attachment) {
        return attachment.c();
    }

    public static long getSize(MessageAttachment attachment) {
        return attachment.d();
    }

    public static MessageAttachmentType getType(MessageAttachment attachment) {
        return attachment.e();
    }

    public static String getUrl(MessageAttachment attachment) {
        return attachment.f();
    }

    public static Integer getWidth(MessageAttachment attachment) {
        return attachment.g();
    }
}

File: Aliucord/src/main/java/com/aliucord/wrappers/messages/AttachmentWrapper.kt
/*
 * This file is part of Aliucord, an Android Discord client mod.
 * Copyright (c) 2021 Juby210 & Vendicated
 * Licensed under the Open Software License version 3.0
 */

package com.aliucord.wrappers.messages

import com.discord.api.message.attachment.MessageAttachment
import com.discord.api.message.attachment.MessageAttachmentType

/**
 * Wraps the obfuscated [MessageAttachment] class to provide nice method names and require only one central
 * update if method names change after an update
 */
@Suppress("unused")
class AttachmentWrapper(private val attachment: MessageAttachment) {
  /** Returns the raw (obfuscated) [MessageAttachment] Object associated with this wrapper */
  fun raw() = attachment

  val filename
    get() = attachment.filename

  val height
    get() = attachment.height

  val proxyUrl
    get() = attachment.proxyUrl

  val size
    get() = attachment.size

  val type
    get() = attachment.type

  val url
    get() = attachment.url

  val width
    get() = attachment.width

  companion object {
    @JvmStatic
    val MessageAttachment.filename: String
      get() = a()

    @JvmStatic
    val MessageAttachment.height: Int?
      get() = b()

    @JvmStatic
    val MessageAttachment.proxyUrl: String
      get() = c()

    @JvmStatic
    val MessageAttachment.size: Long
      get() = d()

    @JvmStatic
    val MessageAttachment.type: MessageAttachmentType
      get() = e()

    @JvmStatic
    val MessageAttachment.url: String
      get() = f()

    @JvmStatic
    val MessageAttachment.width: Int?
      get() = g()
  }
}
